jakttest/jakttest.jakt

```diff
@@ -10,113 +10,125 @@ import parser { Parser }
-        // Whether the file is executable by the current user
-        function is_executable(this) -> bool
-        function is_regular_file(this) -> bool
-        function is_directory(this) -> bool
-    }
+import extern "fs.h" {
+    namespace fs {
+        // subset of `struct stat` from <sys/stat.h> that contains
+        // just what is used.
+        extern struct StatResults {
+            function modified_time(this) -> usize
+            // Whether the file is executable by the current user
+            function is_executable(this) -> bool
+            function is_regular_file(this) -> bool
+            function is_directory(this) -> bool
+        }
 
-    extern class DirectoryIterator {
-        // Get next listing name. Note that the full path is not given,
-        // just the base name.
-        public function next(mut this) throws -> String?
-    }
-    // Get a listing of a directory, using readdir().
-    extern function list_directory(path: String) throws -> DirectoryIterator
-    // Returns whether the given path is a directory
+        extern class DirectoryIterator {
+            // Get next listing name. Note that the full path is not given,
+            // just the base name.
+            public function next(mut this) throws -> String?
+        }
+        // Get a listing of a directory, using readdir().
+        extern function list_directory(path: String) throws -> DirectoryIterator
+        // Returns whether the given path is a directory
 
-    // Create a directory
-    extern function mkdir(path: String) throws
-    
-    // Remove a directory
-    extern function rmdir(path: String) throws
+        // Create a directory
+        extern function mkdir(path: String) throws
+        
+        // Remove a directory
+        extern function rmdir(path: String) throws
 
-    // Remove a file
-    extern function unlink(path: String) throws
+        // Remove a file
+        extern function unlink(path: String) throws
 
-    // Calls `stat` on a file. Silences `ENOENT` error by setting
-    // StatResults.exists to false.
-    extern function stat_silencing_enoent(path: String) throws -> StatResults?
+        // Calls `stat` on a file. Silences `ENOENT` error by setting
+        // StatResults.exists to false.
+        extern function stat_silencing_enoent(path: String) throws -> StatResults?
 
 
-    function is_directory(path: String) throws -> bool {
-        let stat = stat_silencing_enoent(path)
-        return stat.has_value() and stat!.is_directory()
     }
+}
 
-    function mkdir_if_not_present(path: String) throws {
-        let stat = stat_silencing_enoent(path)
-        if not stat.has_value() or not stat!.is_directory() {
-            fs::mkdir(path)
-        }
+// FIXME: These three functions should be under a `namespace fs`, but
+// introducing this inside a namespace seems to override the namespace from
+// import extern (which shouldn't happen, it should rather add these)
```

Comment:
do we have an issue for this? be good to reference it (either `#1234` or with a link to the issue)

Reply:
no, but I can raise it.